src/wui/inputqueuedisplay.cc

```diff
@@ -19,586 +19,574 @@
 
 #include "wui/inputqueuedisplay.h"
 
-#include "economy/input_queue.h"
-#include "economy/request.h"
-#include "graphic/rendertarget.h"
+#include "base/i18n.h"
 #include "graphic/style_manager.h"
 #include "graphic/text_layout.h"
 #include "logic/player.h"
 #include "wui/interactive_base.h"
 
-static char const* pic_priority_low = "images/wui/buildings/low_priority_button.png";
-static char const* pic_priority_normal = "images/wui/buildings/normal_priority_button.png";
-static char const* pic_priority_high = "images/wui/buildings/high_priority_button.png";
-static char const* pic_max_fill_indicator = "images/wui/buildings/max_fill_indicator.png";
-
-InputQueueDisplay::InputQueueDisplay(UI::Panel* const parent,
-                                     int32_t const x,
-                                     int32_t const y,
-                                     InteractiveBase& ib,
-                                     Widelands::Building& building,
-                                     const Widelands::InputQueue& queue,
-                                     bool no_capacity_buttons,
-                                     bool no_priority_buttons)
-   : UI::Panel(parent, x, y, 0, 28),
-     interactive_base_(ib),
-     building_(building),
-     queue_(&queue),
-     settings_(nullptr),
-     priority_radiogroup_(nullptr),
-     increase_max_fill_(nullptr),
-     decrease_max_fill_(nullptr),
-     increase_real_fill_(nullptr),
-     decrease_real_fill_(nullptr),
-     index_(queue.get_index()),
-     type_(queue.get_type()),
-     max_fill_indicator_(g_image_cache->get(pic_max_fill_indicator)),
-     cache_size_(queue.get_max_size()),
-     cache_max_fill_(queue.get_max_fill()),
-     total_height_(0),
-     no_capacity_buttons_(no_capacity_buttons),
-     no_priority_buttons_(no_priority_buttons) {
-	if (type_ == Widelands::wwWARE) {
-		const Widelands::WareDescr& ware = *queue.owner().tribe().get_ware_descr(queue_->get_index());
-		set_tooltip(ware.descname().c_str());
-		icon_ = ware.icon();
-	} else {
-		const Widelands::WorkerDescr& worker =
-		   *queue.owner().tribe().get_worker_descr(queue_->get_index());
-		set_tooltip(worker.descname().c_str());
-		icon_ = worker.icon();
+constexpr int8_t kButtonSize = 25;
+
+// The slider indexes the available priorities as 0..4, so here are some conversion functions
+static size_t priority_to_index(const Widelands::WarePriority& priority) {
+	if (priority == Widelands::WarePriority::kVeryLow) {
```

Comment:
`switch(priority)` is more readable + more efficient, like you did in the functions below.

Reply:
Not possible. `WarePriority` would need to get an implicit `int` conversion operator to be usable in `switch`, which I don't want for type-safety.
https://stackoverflow.com/questions/4535825/error-switch-quantity-not-an-integer#4535852

Reply:
Duh, of course - it's a class.